code/modules/mob/living/simple_animal/friendly/nian_caterpillar.dm

```diff
@@ -0,0 +1,187 @@
+	for(var/mob/living/carbon/human/H in C.contents)
+		H.remove_status_effect(STATUS_EFFECT_COCOONED)
+		H.remove_status_effect(STATUS_EFFECT_BURNT_WINGS)
+	C.preparing_to_emerge = FALSE
+	qdel(C)
+
+/datum/action/innate/nian_caterpillar/evolve/Activate()
+
+	var/mob/living/simple_animal/nian_caterpillar/user = owner
+
+	user.visible_message("<span class='notice'>[user] begins to hold still and concentrate on weaving a cocoon...</span>", "<span class='notice'>You begin to focus on weaving a cocoon... (This will take [COCOON_WEAVE_DELAY / 10] seconds, and you must hold still.)</span>")
+	if(do_after(user, COCOON_WEAVE_DELAY, FALSE, user))
+		var/obj/structure/moth/cocoon/C = new(get_turf(user))
+		var/datum/mind/H = user.mind
+		user.evolve(C, H)
+		addtimer(CALLBACK(src, PROC_REF(emerge), C), COCOON_EMERGE_DELAY, TIMER_UNIQUE)
+
+/mob/living/simple_animal/nian_caterpillar/proc/evolve(obj/structure/moth/cocoon/C, datum/mind/M)
+	if(stat != CONSCIOUS)
+		return FALSE
+
+	// A changeling caterpillar shouldn't be restricted from evolving.
+	// A caterpillar needs to consume food-- similar to a dioan nymph --to evolve.
+	if((nutrition < nutrition_need) && !IS_CHANGELING(M))
+		to_chat(src, "<span class='warning'>You need to binge on food in order to have the energy to evolve...</span>")
+		src.health = src.maxHealth // Heal the caterpillar for cocooning.
+		src.KnockOut()
+		qdel(C)
+		return FALSE
+
+	// Worme is the lesser form of nian. The caterpillar evolves into this lesser form.
+	var/mob/living/carbon/human/nian_worme/adult = new(get_turf(loc))
+
+	if(istype(loc, /obj/item/holder/nian_caterpillar))
+		var/obj/item/holder/nian_caterpillar/L = loc
+		forceMove(L.loc)
+		qdel(L)
+
+	for(var/datum/language/L in languages)
+		adult.add_language(L.name)
+	adult.regenerate_icons()
+	adult.name = src.name
+	adult.real_name = adult.name
+
+	// Mind transfer to new worme.
+	M.transfer_to(adult)
+
+	// Worme is placed into cacoon.
+	adult.forceMove(C)
+	C.preparing_to_emerge = TRUE
+	adult.apply_status_effect(STATUS_EFFECT_COCOONED)
+	adult.KnockOut() // Zzzz
+	adult.create_log(MISC_LOG, "has woven a cocoon")
+
+	// For any random generated names. This is for when a new nian caterpillar is spawned.
+	// [ nian caterpillar (042) ] etc.
+	if(findtext(adult.real_name, "nian caterpillar") != 0)
+		adult.real_name = adult.dna.species.get_random_name()
+		adult.name = adult.real_name
+
+	for(var/obj/item/W in contents)
+		unEquip(W)
+
+	qdel(src)
+	return TRUE
+
+/mob/living/simple_animal/nian_caterpillar/proc/consume(obj/item/food/snacks/G)
+	if(nutrition >= nutrition_need) // Prevents griefing by overeating food items without evolving.
+		to_chat(src, "<span class='warning'>You're too full to consume this! Perhaps it's time to grow bigger...</span>")
+	else
+		visible_message("[src] ravenously consumes [G].", "You ravenously devour [G].")
+		playsound(loc, 'sound/items/eatfood.ogg', 30, 0, frequency = 1.5)
```

Comment:
```suggestion
		playsound(loc, 'sound/items/eatfood.ogg', 30, TRUE, frequency = 1.5)
```
Standard to use vary on so its not the same exact sound over and over. However if theres a reason to have if off, disregard the change and just replace the `0` with `FALSE`